aws-creds/iam.tf

```diff
@@ -59,7 +66,64 @@
 
 #data "aws_caller_identity" "current" {}
 
+resource "aws_iam_role" "terraform_architect_iam_role" {
+  name = "terraform-architect"
+  path = "/"
+  assume_role_policy = <<EOF
```

Comment:
Can you use an `aws_iam_policy_document` here instead of just using JSON directly? See [this](https://github.com/yuvipanda/terraform-deploy/blob/fix/no-users/aws/iam.tf#L30)